v0.11.0 Upgrade Documentation (#4525)
hashicorp/waypoint: website/content/docs/upgrading/version-guides/0.11.0.mdx

---
layout: docs
page_title: Version 0.11.0 - Upgrade Guide
description: |-

>   This upgrade guide describes any considerations for upgrading to Waypoint 0.11.0.

Reviewer: ```suggestion
  This upgrade guide describes considerations for upgrading to Waypoint 0.11.0.
```

---

# Upgrading to Waypoint 0.11.0 from 0.10.x

Waypoint 0.10.x can be upgraded to Waypoint 0.11.0 using a [standard upgrade](/waypoint/docs/upgrading#standard-upgrade).
There are some considerations depending on what platform your server is installed on.

> There are no breaking changes in the protocol or `waypoint.hcl` configuration file.

Author: Thank you for articulating this - I almost omitted this sentence in my draft, but since neither the protocol nor the config file have breaking changes, I did leave it and left the issue to be discussed below, but I like that this calls it out earlier.

> Due to the changes in `runner` management, we cannot guarantee that all Waypoint
> components (CLI, entrypoint, and server) are forwards and backwards
> compatible. We encourage users to have all components on the same version.

Reviewer: This was true for the 0.9 upgrade, but is not true for 0.11.
```suggestion
```

if there is a dynamic configuration source set using the `waypoint config
source-set` command, like so.

If your server is running at either of these versions and a config
source is set, then you will not be able to take snapshots using the command
`server snapshot`. By default the `server upgrade` command will attempt to take
a snapshot of the database before upgrading and this will fail due to this bug.

There are a few options to upgrade to v0.11.0. If you do not need a snapshot of
the database before upgrading, then you may run `server upgrade` with the flag
`-snapshot=false` to proceed with upgrading, like so.

```shell
$ waypoint server upgrade -snapshot=false
```

Another alternative is to delete all config sources from Waypoint using
`config source-set` with the `-delete` flag, and then running `server upgrade`.
After the server is upgraded, you can re-create the config sources using
`config source-set`.

```shell
# Repeat the command below for each config source that is set; this example
# deletes the globally-scoped Vault config source
$ waypoint config source-set -delete -type=vault -scope=global

> ````

Reviewer: there's an extra tic mark here. I can't make a suggestion because suggestions use tic marks.

A third approach is to use platform-native persistence solutions to back up the
file system where the Waypoint database is stored. This includes Docker, AWS
ECS, Kubernetes, and HashiCorp Nomad. The steps to do this with each of these
platforms are detailed below.

### Docker

The Waypoint server stores data into a single `data.db` file, in Docker it's
normally stored in `/data/data.db`. In the server data directory, copy the
snapshot data and name the file `waypoint-restore.db`.

> Now run `waypoint server upgrade -snapshot=false`.

Reviewer: ```suggestion
Now run `waypoint server upgrade -snapshot=false -platform=docker`.
```

Reviewer: Nevermind, we default now! Cool!

> Now run `waypoint server upgrade -snapshot=false`.

Reviewer: For consistency, can we do this here and for the following 4 commands?
```suggestion
Now run:
```shell
 $ waypoint server upgrade -snapshot=false
tic marks * 3
```

---
layout: docs
page_title: Version 0.11.0 - Upgrade Guide
description: |-
  This upgrade guide describes considerations for upgrading to Waypoint 0.11.0.
---

# Upgrading to Waypoint 0.11.0 from 0.10.x

Waypoint 0.10.x can be upgraded to Waypoint 0.11.0 using a [standard upgrade](/waypoint/docs/upgrading#standard-upgrade).
There are some considerations depending on what platform your server is installed on.

> There are no breaking changes in the protocol or `waypoint.hcl` configuration file. However, there was a bug related to the usage of dynamic configuration which will require attention during the upgrade. Those details are below.

Author: ```suggestion
There are no breaking changes in the protocol or `waypoint.hcl` configuration file. 
However, there was a bug related to the usage of dynamic configuration which will require attention during the upgrade. Those details are below.
```

## Server Snapshot

Waypoint servers at versions 0.10.4 and 0.10.5 have a bug that prevents
snapshots from being taken if a dynamic configuration source was set using the
`waypoint config source-set` command. If your server is running at either of
these versions and a `waypoint config source-set` command was run, then you
will not be able to take snapshots using the command `waypoint server snapshot`.

> When upgrading to Waypoint 0.11.0, follow your normal upgrade procedures. To

Reviewer: ```suggestion
To
```

check if you are affected by the bug in question, run
`waypoint server snapshot`. If this succeeds, disregard the rest of the below
remediation steps. You are unaffected by this bug and can use
`waypoint server upgrade` with no extra steps beyond your normal upgrade

> procedure. If this fails, proceed with the below remediation steps. There are

Reviewer: ```suggestion
procedure.

If this fails, proceed with the below remediation steps. There are
```

three options to upgrade to v0.11.0 if you are unable to create snapshots.

> The first is, you do not need a snapshot of the database before upgrading, then

Reviewer: ```suggestion

The first is, you do not need a snapshot of the database before upgrading, then
```

> The first is, you do not need a snapshot of the database before upgrading, then

Author: ```suggestion
The first is, if you do not need a snapshot of the database before upgrading, then
```

> run server upgrade with the flag `-snapshot=false` to proceed with upgrading:

Reviewer: ```suggestion
run the server upgrade with the flag `-snapshot=false` to proceed with upgrading:
```

> run server upgrade with the flag `-snapshot=false` to proceed with upgrading:

Author: ```suggestion
run `waypoint server upgrade` with the flag `-snapshot=false` to proceed with upgrading:
```

```shell
$ waypoint server upgrade -snapshot=false
````

The second alternative is to delete all config sources from Waypoint using
`waypoint config source-set` with the `-delete` flag, and then run
`waypoint server upgrade`. After the server is upgraded, you can re-create the
config sources using `waypoint config source-set`.

```shell
# Repeat the command below for each config source that is set; this example
# deletes the globally-scoped Vault config source
waypoint config source-set -delete -type=vault -scope=global
```

The third approach is to use platform-native persistence solutions to back up
the file system where the Waypoint database is stored. This includes Docker,
AWS ECS, Kubernetes, and HashiCorp Nomad. The steps to do this with each of
these platforms are detailed below.

### Docker

> The Waypoint server stores data into a single `data.db` file, in Docker it's

Reviewer: ```suggestion
The Waypoint server stores data in a single `data.db` file, in Docker it's
```

> normally stored in `/data/data.db`. In the server data directory, copy the

Reviewer: ```suggestion
default location is `/data/data.db`. In the server data directory, copy the
```

---
layout: docs
page_title: Version 0.11.0 - Upgrade Guide
description: |-
This upgrade guide describes any considerations for upgrading to Waypoint 0.11.0.
---

# Upgrading to Waypoint 0.11.0 from 0.10.x

Waypoint 0.10.x can be upgraded to Waypoint 0.11.0 using a [standard upgrade](/waypoint/docs/upgrading#standard-upgrade).
There are some considerations depending on what platform your server is installed on.
There are no breaking changes in the protocol or `waypoint.hcl` configuration file.
Due to the changes in `runner` management, we cannot guarantee that all Waypoint
components (CLI, entrypoint, and server) are forwards and backwards
compatible. We encourage users to have all components on the same version.

## Server Snapshot

> Waypoint servers at versions 0.10.4 and 0.10.5 have a bug with taking snapshots,

Reviewer 2: Just a thought, could be neat to link to the bug fix PR here! Not required though.
```suggestion
Waypoint servers at versions 0.10.4 and 0.10.5 have a [bug with taking snapshots](https://github.com/hashicorp/waypoint/pull/4523),
```

Author: I agree!

if there is a dynamic configuration source set, using the `config source-set`
command. If your server is running at either of these versions and a config
source is set, then you will not be able to take snapshots using the command
`server snapshot`. By default, the `server upgrade` command will also take a
snapshot of the database, and this also fails due to this bug.

There are a few options to upgrade to v0.11.0. If you do not need a snapshot of
the database before upgrading, then you may run `server upgrade` with the flag
`-snapshot=false` to proceed without issue.

Another alternative is to delete all config sources in your database using
`config source-set` with the `-delete` flag, and then running `server upgrade`.
After the server is upgraded, you can re-create the config sources using

> `config source-set`.

Reviewer 2: I'd recommend showing the full command in a code block with an example source-set here so it's easy at a glance to see what people will have to do to fix these configs.

A third approach is to use platform-native persistence solutions to back up the
file system where the Waypoint database is stored. This includes Docker, AWS
ECS, Kubernetes, and HashiCorp Nomad. The steps to do this with each of these
platforms are detailed below.

### Docker

### AWS ECS

The database of a Waypoint server installed to AWS ECS is persisted in a file
system, provided by the Elastic File System (EFS) service of AWS. To preserve
the database during the upgrade, a new EFS volume will need to be created. An
access point will need to be added to this volume as well. This access point's
path must be set to `/waypointserverdata`. The POSIX user and root directory
creation permissions should both have their user and group ID set to `100` and
`1000`, respectively. The access point permissions should be set to `755`.

The data in the old file system must be synchronized to the new one. This may be
done using [AWS DataSync](https://docs.aws.amazon.com/datasync/latest/userguide/what-is-datasync.html).
The source location should be configured as your old file system for the
Waypoint server and the destination should be configured as the file system you
just created.

> Now that there are two file systems in EFS with the in your Waypoint database,

Reviewer 3: Is there a small typo here?

the next step is to upgrade the Waypoint server container to v0.11.0. To do
this, the task definition `waypoint-server` in ECS needs a new revision. The
only setting that must be changed is the image URI, which should be set to
`hashicorp/waypoint:0.11.0`. After the new revision of the task definition is
created, the `waypoint-server` service in ECS must be updated to use the
revision you just created.

If it is necessary to restore the snapshot to Waypoint, then a new revision of
the ECS task definition `waypoint-server` needs to be created. In this new
version, the storage configuration must be updated to replace the existing file
system with the one created as part of this snapshot process. With that updated,
the `waypoint-server` ECS service needs to be updated once more to use the
newest revision of the `waypoint-server` task definition. At this point, the
server should be using the pre-upgrade snapshot of the database.

### Kubernetes

> ### Nomad

Reviewer 2: ```suggestion
### HashiCorp Nomad
```
